### PR TITLE
Support Or queries in Criteria

### PR DESCRIPTION
**Pull request by mbhnyc**
sometimes you’d like to query against multiple fields, now you can!

`client.retrieve_multiple('account', [["telephone1", "EndsWith", "5558675309"], ["mobilephone", "EndsWith", "5558675309"]], [], "Or”)`

### CODE REVIEW

**Comment by mbhnyc** [comment]
as promised!

**Comment by mbhnyc** [comment]
ping @jheth

**Comment by mbhnyc** [comment]
I'd like to get back onto master instead of my fork :)

**Comment by jheth** [comment]
Hey @mbhnyc , sorry for the delay. Thanks for the contribution. I'll take a look.

File: spec/lib/xml/query_spec.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
       it { expect(subject.to_xml).to include("<b:AllColumns>true</b:AllColumns>") }
       it { expect(subject.to_xml).to include("<b:Distinct>false</b:Distinct>") }
       it { expect(subject.to_xml).to include("<b:EntityName>opportunity</b:EntityName>") }
-      it { expect(subject.to_xml).to include("<b:FilterOperator>And</b:FilterOperator>") }
+      it { expect(subject.to_xml).to include("<b:FilterOperator>And</b:FilterOperator>").or(include("<b:FilterOperator>Or</b:FilterOperator>")) }
```
Comment (jheth) [inline spec/lib/xml/query_spec.rb]: Why was this expectation changed?  This spec should always return `And` right?

File: lib/dynamics_crm/client.rb [hunk 2]
```diff
@@ -141,11 +141,10 @@ def rollup(target_entity, query, rollup_type="Related")
         })
     end
 
-    def retrieve_multiple(entity_name, criteria=[], columns=[])
-
+    def retrieve_multiple(entity_name, criteria=[], columns=[], operator=nil)
```
Comment (jheth) [inline lib/dynamics_crm/client.rb]: This method signature is getting a little complicated. I may want to refactor this to named parameters or an object that can wrap up all the search criteria.

File: lib/dynamics_crm/xml/criteria.rb [hunk 3]
```diff
@@ -2,11 +2,14 @@ module DynamicsCRM
   module XML
 
     class Criteria < Array
+      SUPPORTED_OPERATORS = %w(And Or)
 
       attr_accessor :filter_operator
-      def initialize(tuples=[])
-        super
-        @filter_operator = 'And'
+      def initialize(tuples=[], filter_operator=nil)
```
Comment (jheth) [inline lib/dynamics_crm/xml/criteria.rb]: Can we make this a named parameter and should it default to `And`? 
`def initialize(tuples=[], filter_operator: 'And')`

File: spec/lib/client_spec.rb [hunk 4]
```diff
@@ -110,6 +110,15 @@
       expect(entities[1].attributes["accountid"]).to eq("dbe9d7c9-2c98-e311-9752-6c3be5a87df0")
       expect(entities[2].attributes["accountid"]).to eq("8ff0325c-a592-e311-b7f3-6c3be5a8a0c8")
     end
+
+    it "retrieves multiple entities by criteria using OR" do
+
+      allow(subject).to receive(:post).and_return(fixture("retrieve_multiple_result"))
+
+      result = subject.retrieve_multiple("account", ["name", "Equal", "Test Account", 'Or'], columns=[])
```
Comment (jheth) [inline spec/lib/client_spec.rb]: This is the wrong signature for the method. `Or` should come after columns.
`result = subject.retrieve_multiple("account", ["name", "Equal", "Test Account"], columns=[], 'Or')`

**Comment by jheth** [comment]
@mbhnyc I pushed some changes today that caused conflicts. I extracted a ConditionExpression class that I think will be helpful.

**Comment by mbhnyc** [comment]
nice! and yes, definitely time extract some of the complexity, i'll review

**Pull request closed without merging**